binder/src/main/java/io/grpc/binder/internal/ActiveTransportTracker.java

```diff
@@ -0,0 +1,114 @@
+package io.grpc.binder.internal;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.HashSet;
+import javax.annotation.concurrent.GuardedBy;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.internal.ServerStream;
+import io.grpc.internal.ServerTransportListener;
+
+/**
+ * Tracks which {@link BinderTransport.BinderServerTransport} are currently active and allows
+ * invoking a {@link BinderTransportSecurity.TerminationListener} only once all transports are
+ * terminated.
+ */
+final class ActiveTransportTracker {
+  @GuardedBy("this")
+  private final HashSet<BinderTransport.BinderServerTransport> activeTransports = new HashSet<>();
+  private final BinderTransportSecurity.TerminationListener transportSecurityTerminationListener;
+  @GuardedBy("this")
+  private boolean awaitingNoActiveTransports = false;
```

Comment:
Do you need this ? If you prohibit tracking new transports when `shutdown` and assume that callers obey the contract of `ServerTransportListener#transportTerminated` is just `shutdown` enough?

Reply:
I think I had indeed added a separate field to make the class 100% safe from misuse, but it may be overkill indeed. Updated.